### ISSUE TITLE
How to regist custom pytorch layer in apex

### ISSUE THREAD

**Issue by GH-HOME** [NVIDIA/apex#345]
Hi, I want to implement a custom layer to scale the gradient backward process. Here is my layer:
`
import torch

class Rebalance_Op(torch.autograd.Function):
    @staticmethod
    def forward(ctx, input, factor):
        ctx.save_for_backward(input, factor)
        return input * torch.tensor(1.0).cuda()

    @staticmethod
    def backward(ctx, grad_output):
        input, factor = ctx.saved_variables
        grad_input = grad_output * factor
        return grad_input, None
`
If I want to add apex support, I have to add property "@amp.half_function" to both 'forward' and 'backward' function in my Rebalance_Op class. But I already have "@staticmethod" property. How can I do this.

In the current [https://github.com/NVIDIA/apex/tree/master/apex/amp#annotating-user-functions](document ), only custom function is supported by apex, how about custom class?

**Comment by mcarilli** [comment]
Why do you need to register these as half_functions?  I think RebalanceOp should be fine in either fp16 or fp32 (unless `factor` is very large, in which case Rebalance_Op's functions should be float_functions rather than half_functions).  In other words, don't worry about registering Rebalance_Op with Amp at all, in which case Rebalance_Op will simply process inputs in whatever type it receives.

**Comment by YuwenXiong** [comment]
@mcarilli I believe @GH-HOME just gave a general case. I also have the same problem, my operator will call custom cuda kernel which only support float32 type. Do you have any idea about how to disable casting for specific custom operator with static forward/backward method?
